optimizations.md

```diff
@@ -0,0 +1,9 @@
+# Client-side optimizations
+
+This document reflects client-side optimizations that are implemented in this repository. Client-side optimizations are not part of the [Kademlia spec](https://github.com/libp2p/specs/tree/master/kad-dht), and are not required to be implemented on all clients.
+
+## Checking before Adding
+
+Owner: [Guillaume Michel](https://github.com/guillaumemichel)
```

Comment:
Should this be covered by the `CODEOWNER` file ? No one reads markdown anyway :upside_down_face: